src/classifier.py

```diff
@@ -1,30 +1,87 @@
 import torch
 import torch.nn as nn
+import torch.optim as optim
+from torch.utils.data import TensorDataset, DataLoader
+
+import numpy as np
+
+import spacy
+
 from collections import Counter
 import re
-import torch.optim as optim
+import os 
+from time import time
 torch.manual_seed(1)
 

```

Comment:
Woow, this is a long file with so many classes! Maybe you could split this into multiple scripts. I know, the preprocessing step depends on the model but I think it would be more "modular" if there were a module for each step of the pipeline. For example, there could be one module for preprocessing sentences and everybody could call methods from it that will be different or similar depending on the model. 
An example of arborescence : 
- classifier.py : classifier + encoder
- preprocessing.py : SentencePreprocessor + Dataset

This script is also very long to read, it would be easier to understand it if each task was split in my opinion.